dandischema/metadata.py

```diff
@@ -158,16 +164,20 @@ def migrate(
         id = obj.get("id")
         if not id.startswith("DANDI:"):
             obj["id"] = f'DANDI:{obj["id"]}'
-        for contrib in obj["contributor"]:
+        for contrib in obj.get("contributor", []):
+            if not contrib.get("roleName"):
+                continue
             contrib["roleName"] = [
                 val.replace("dandi:", "dcite:") for val in contrib["roleName"]
             ]
```

Comment:
```suggestion
            if contrib.get("roleName"):
                contrib["roleName"] = [
                    val.replace("dandi:", "dcite:") for val in contrib["roleName"]
                ]
```
since otherwise we would not add schemaKey for affiliation below for contribs without role